server/game/stateWatchers/AttacksThisPhaseWatcher.ts

```diff
@@ -57,6 +57,7 @@ export class AttacksThisPhaseWatcher extends StateWatcher<IAttacksThisPhase> {
         filter?: (event: AttackEntry) => boolean;
     }) {
         return this.getAttackers((entry) => {
+            // eslint-disable-next-line eqeqeq
```

Comment:
Huh, really weird that this happened, usually it's okay with this type of check. Could you try doing it like this and see if that lets us remove the eslint comment (we try to avoid doing that if possible):

```ts
const additionalFilter = filter ? filter(entry) : true;
```

Reply:
alright, I updated to change this. I also was thinking, would it be better to check `typeof filter === function` since it needs to be a function to not error?